File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Classes/StartingPosition.java
package com.alphadevelopmentsolutions.frcscout.Classes;

/**
 * Positions available before a match
 */
public enum StartingPosition
{
    LEFT,
    CENTER,
    RIGHT;

    /**
     * Converts a string value to enum
     * @param position string enum name
     * @return enum converted from string
     */
    public static StartingPosition getPositionFromString(String position)
    {
        if(position.toLowerCase().equals(StartingPosition.LEFT.name().toLowerCase()))
            return StartingPosition.LEFT;
        else if(position.toLowerCase().equals(StartingPosition.CENTER.name().toLowerCase()))
            return StartingPosition.CENTER;
        else
            return StartingPosition.RIGHT;
    }
}

File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Enums/StartingPosition.kt
package com.alphadevelopmentsolutions.frcscout.Enums

/**
 * Positions available before a match
 */
enum class StartingPosition
{
    LEFT,
    CENTER,
    RIGHT;


    companion object
    {

        /**
         * Converts a string value to enum
         * @param position string enum name
         * @return enum converted from string
         */
        fun getPositionFromString(position: String): StartingPosition
        {
            return when
            {
                position.toLowerCase() == StartingPosition.LEFT.name.toLowerCase() -> LEFT
                position.toLowerCase() == StartingPosition.CENTER.name.toLowerCase() -> CENTER
                else -> RIGHT
            }
        }
    }
}
